docs/user/README.md

```diff
@@ -200,6 +200,25 @@ cass-operator. Within a single Kubernetes namespace:
 For this guide, we define a single-datacenter cluster. The cluster is named
 `cluster1` with the datacenter named `dc1`.
 
+## Datacenter name
+
+By default, datacenters will be named after the `metadata.name` of the CassandraDatacenter object. This can be overridden by setting the `spec.datacenterName` field. This is useful when you want to use a different name for the datacenter than the name of the object.
+
+```yaml
+apiVersion: cassandra.datastax.com/v1beta1
+kind: CassandraDatacenter
+metadata:
+  name: cluster1-dc1
+spec:
+  clusterName: cluster1
+  datacenterName: dc1
+  serverType: cassandra
+  serverVersion: 3.11.7
+  ...
+```
+
+Objects such as statefulsets, pods, secrets will be named and labeled after the `spec.datacenterName` value. This value can contain characters which aren't compliant with Kubernetes naming rules (such as capital letters or underscores) and will be sanitized when used in object names/labels. For example, `Datacenter_1` will be sanitized to `datacenter-1`.
```

Comment:
We should not do this. The name of the Kubernetes objects should follow the name of the ``CassandraDatacenter``.

For a very simple reason also, imagine I create two CassandraDatacenters:

```yaml
kind: CassandraDatacenter
metadata:
  name: dc1
```

```yaml
kind: CassandraDatacenter
metadata:
  name: cluster1-dc1
spec:
  datacenterName: dc1
```

Kaboom! Now these two CassandraDatacenters can't suddenly be queried and they will overlap since both have ``dc1`` everywhere. They need to follow the name of the CassDc to avoid any conflicts.

Reply:
I've added checks for this now. We can detect any conflict with another cassdc.